openml/tasks/functions.py

```diff
@@ -292,9 +292,12 @@ def get_task(task_id):
         try:
             task = _get_task_description(task_id)
             dataset = get_dataset(task.dataset_id)
-            class_labels = dataset.retrieve_class_labels(task.target_name)
-            task.class_labels = class_labels
-            task.download_split()
+            # Clustering tasks do not have class labels
+            # and do not offer download_split
+            if not isinstance(task, OpenMLClusteringTask):
+                task.class_labels = \
+                    dataset.retrieve_class_labels(task.target_name)
+                task.download_split()
```

Comment:
Would it make sense to move this into the task classes? If they have a split and class labels they retrieve them, otherwise, they don't. Also, how does this work for regression tasks (regarding class labels)?

Reply:
Nice point, it does not work for regression. What do you think about this

```
if isinstance(task, OpenMLSupervisedTask):
    task.download_split()
    if isinstance(task, OpenMLClassificationTask): 
        task.class_labels = \
            dataset.retrieve_class_labels(task.target_name)

```

Reply:
Agreed. Your proposed solution is fine for this.